hs_core/management/commands/list_published_by_year.py

```diff
@@ -116,3 +146,41 @@ def print_resource(self, res, pub_date):
                     print("No award number")
         else:
             print("Resource has no funding information")
+
+    def build_csv(self, res, pub_date, site_url):
+        row = []
+        res_url = site_url + res.absolute_url
+        funding_agencies = res.metadata.funding_agencies.all()
+        row.append(res_url if res_url else None)
+        row.append(res.doi if res.doi else None)
+        title = res.metadata.title.value
+        row.append(title if title else None)
+        row.append(res.resource_type if res.resource_type else None)
+        row.append(pub_date if pub_date else None)
+
+        if funding_agencies:
+            num_funders = len(funding_agencies)
+            if num_funders > self.max_funders:
+                self.max_funders = num_funders
+            for f in funding_agencies:
+                row.append(f.agency_name if f.agency_name else None)
+                row.append(f.agency_url if f.agency_url else None)
+                row.append(f.award_title if f.award_title else None)
+                row.append(f.award_number if f.award_number else None)
+        self.rows.append(row)
+
+    def write_csv(self, writer):
+        # write headers last since funders is variable number
+        headers = ['url', 'doi', 'title', 'type', 'pub_date']
+        for i in range(1, self.max_funders + 1):
+            headers.append(f"agency_name_{i}")
+            headers.append(f"agency_url_{i}")
+            headers.append(f"award_title_{i}")
+            headers.append(f"award_number_{i}")
+
+        # insert the headers
+        writer.writerow(headers)
```

Comment:
This function looks like doesn't need the `writer` parameter.
`self.writer.writerow(headers)`

Reply:
yep, thanks, changed that